### PR TITLE
dom: Worker entity initial commit (wip)

### PR DESCRIPTION
**Pull request by vlkolmakov**
This PR starts a series of small, granular implementations for the OME server domain object model from the ground up.

Instead of huge, hard to grasp commits, these will be per entity pull requests with the following workflow:

1. One or several related entity definitions for open discussion
2. Database storage
3. Parsing from the Job Description Language

@jasonprado please review a99e551073929aa7a7ee8199cb1c3d3e64b3dba5. I need your approval of the Worker model to move forward with database storage.

@kepkap I've disabled whitespace linter together with auto-fix. Don't want some stupid machinery to mess with my code style.

### CODE REVIEW

File: dom/worker.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	WorkerCreated   = iota // a worker registered in the system
+	WorkerApproval         // worker is pending regulatory/system approval
+	WorkerOffline          // worker is offline
+	WorkerReady            // worker is ready for a job
+	WorkerOnjob            // worker is on the job
+	WorkerSuspended        // worker is suspended for some reason
+	WorkerRetired          // worker is retired and will no longer be available
+)
+
+// Worker represents information about a driver.
```
Comment (kepkap) [inline dom/worker.go:19]: review feedback:
id
Status
Rank => Rating
Vehicle

Comment (kepkap) [reply dom/worker.go:19]: first name last name

File: dom/worker.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 The Drivers Cooperative. All rights reserved.
+// Use of this source code is governed by a dual
+// license that can be found in the LICENSE file.
+
+package dom
+
+type WorkerStatus int32
+
+const (
```
Comment (jasonprado) [inline dom/worker.go:9]: We need:
- Offline
- Ready
- OnJob
- Paused

File: dom/worker.go (hunk 1 above)
Comment (kepkap) [reply dom/worker.go:19]: first name
last name

Comment (kepkap) [reply dom/worker.go:19]: first name last name

**Comment by vlkolmakov** [comment]
I need a couple of Time fields to test proper database date/time handling. So I keep `Created`, `Updated`.

File: .golangci.yml [hunk 3]
```diff
@@ -63,11 +63,11 @@ linters:
     - unconvert
     - unparam
     - varcheck
-    - whitespace
+#    - whitespace
 #    - wsl
 
 issues:
-  fix: true
+  fix: false
```
Comment (jasonprado) [inline .golangci.yml:70]: I think we should leave `fix` on, and regardless it should be changed in its own PR if it's to be changed. Same for rules above.

File: cfg/pgdb.go [hunk 4]
```diff
@@ -39,13 +37,23 @@ func (c *PgdbConfig) Check(name ...string) error {
 }
 
 // FullAddr appends PgdbConfig.Addr with optional parameters.
-func (c *PgdbConfig) FullAddr() string {
-	const sslmode = "sslmode"
+func (c *PgdbConfig) FullAddr() (string, error) {
 	a := c.Addr
+	if len(a) == 0 {
+		if user := os.Getenv("USER"); len(user) > 0 {
+			a = Address(fmt.Sprintf("postgres://%s@localhost:5432/%s", user, user))
```
Comment (jasonprado) [inline cfg/pgdb.go:44]: This reminds me, we should put postgres in `docker-compose.yml` soon.

**Review by jasonprado — APPROVED** [review]

File: .golangci.yml (hunk 3 above)
Comment (vlkolmakov) [reply .golangci.yml]: I've reverted the linter changes 775ce28. Will do a separate PR for this.

The preliminary arguments to disable both are:
1. Nothing should mess with the code because it is unpredictable what's going to be changed and where. The sole linter task is to issue warnings/errors.
2. Whitespace rules are especially annoying. For some functions I use extra new line after each statement to greatly improve readability. This is not only beneficial for myself, but other team members do have easier time to read/review the critical code paths.
3. The eventual source of truth for Go code formatting is gofmt, and it has not issues with extra new lines where I use it.
